### PR TITLE
Feature/textarea autocomplete

### PR DESCRIPTION
**Pull request by makaroni4**
Added JS for positioning usernames block near '@'

### CODE REVIEW

File: vendor/assets/javascripts/areacomplete.js [hunk 1]
```diff
@@ -35,6 +35,19 @@ jQuery.fn.extend({
   }
 });
 
+function getcaretPosition(ctrl) {
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: getCaretPosition

File: vendor/assets/javascripts/areacomplete.js [hunk 2]
```diff
@@ -195,7 +208,7 @@ $(document).ready(function() {
     }
 
     this.live('keydown', function(e) {
-      var textarea = $(this);
+      var textarea = $(this);
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js:215]: Is it really need? :)

File: vendor/assets/javascripts/areacomplete.js [hunk 3]
```diff
@@ -218,7 +231,71 @@ $(document).ready(function() {
           return true;
         }
 
+        /// ----- calculating autocomplete offset -----
+
+        var offset = getcaretPosition(this);
+        
+        var div = '<div id="post_content_dup" class="wordwrap" style="opacity:0;position:relative;z-index:-10"></div>';
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: To CSS

Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: Add class "invisible" to CSS and use it to div classes

File: vendor/assets/javascripts/areacomplete.js [hunk 4]
```diff
@@ -218,7 +231,71 @@ $(document).ready(function() {
...
+				}
+        				                
+        $dup_div.text(textarea.val());
+
+        var span = '<span class="ins_str">@</span>';
+        $dup_div.html($dup_div.text().substr(0, offset) + span + $dup_div.text().substr(offset, $dup_div.text().length));
+
+        var x = $dup_div.offset().left,
+            y = $dup_div.offset().top,
+            width = $dup_div.outerWidth() - 15,
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: What is 15?

File: vendor/assets/javascripts/areacomplete.js [hunk 5]
```diff
@@ -218,7 +231,71 @@ $(document).ready(function() {
...
+            height = $dup_div.outerHeight() - 15;
+
+        $dup_div.html(function(index, old) {
+          return old.replace(/\n/g, '<br>')
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: `<br />`

File: vendor/assets/javascripts/areacomplete.js [hunk 6]
```diff
@@ -218,7 +231,71 @@ $(document).ready(function() {
...
+          var span_offset = $span.offset(),
+              dup_div_offset = $dup_div.offset(),
+              char_x = span_offset.left - dup_div_offset.left,
+              char_y = span_offset.top - dup_div_offset.top;
+        }
+        
+        // fix for Firefox
+        // auto-resize plugin hides one line so we need to
+        // subtract its height from vertical offset
+        if($.browser.mozilla && textarea.height() > 32 && char_y > 18) {
```
Comment (releu) [inline vendor/assets/javascripts/areacomplete.js]: Also what is 18 and 32?

File: vendor/assets/javascripts/areacomplete.js (hunk 4 above)
Comment (makaroni4) [inline vendor/assets/javascripts/areacomplete.js]: OMG this is some old code needed to be deleted.

File: vendor/assets/javascripts/areacomplete.js (hunk 6 above)
Comment (makaroni4) [inline vendor/assets/javascripts/areacomplete.js]: Actually there is a comment in the next line :)

But I agree, I will move it to variable

File: vendor/assets/javascripts/areacomplete.js (hunk 2 above)
Comment (kirs) [inline vendor/assets/javascripts/areacomplete.js:215]: why you don't use the coffeescript?

Comment (releu) [inline vendor/assets/javascripts/areacomplete.js:215]: we'll

**Pull request merged**